### ISSUE TITLE
error when copying id.json

### ISSUE THREAD

**Issue by fxtrader-react** [dyor-market/mango-v3-service#1]
I guess I'm doing something wrong but when I run docker-compose up. I get error while trying to copy id.json. Would love to find a solution and give this bot a spin

**Comment by fxtrader-react** [comment]
`WARNING: The CLUSTER_URL variable is not set. Defaulting to a blank string.
Building mango-service-v3
[+] Building 1.3s (10/14)                                                       
 => [internal] load build definition from Dockerfile                       0.0s
 => => transferring dockerfile: 37B                                        0.0s
 => [internal] load .dockerignore                                          0.0s
 => => transferring context: 34B                                           0.0s
 => [internal] load metadata for docker.io/library/node:16                 1.1s
 => [internal] load build context                                          0.0s
 => => transferring context: 2.24kB                                        0.0s
 => CANCELED [ 1/10] FROM docker.io/library/node:16@sha256:3afa49c0e36da2  0.1s
 => => resolve docker.io/library/node:16@sha256:3afa49c0e36da219d62460bc3  0.0s
 => => sha256:3afa49c0e36da219d62460bc30fe3da87792392fa96 1.21kB / 1.21kB  0.0s
 => => sha256:44e59f8582c39b17ab895261edbcfeadb082ea2a881 2.21kB / 2.21kB  0.0s
 => => sha256:054bccad10b08705d61ff87ad3ccf825b297a68d13e 7.65kB / 7.65kB  0.0s
 => CACHED [ 2/10] RUN npm install -g pm2                                  0.0s
 => CACHED [ 3/10] WORKDIR /usr                                            0.0s
 => CACHED [ 4/10] COPY package.json ./                                    0.0s
 => CACHED [ 5/10] COPY yarn.lock ./                                       0.0s
 => ERROR [ 6/10] COPY id.json ./ `

**Comment by staccDOTsol** [comment]
hi! i think for the older set of directions, copying from ~/.config/solana/id.json, you'll first need to run solana-keygen new 

iirc

**Comment by staccDOTsol** [comment]
it should work better here: https://github.com/dyor-market/Mango-Money-Multiplier/tree/main

**Comment by fxtrader-react** [comment]
Ok thanks, I got the solana keygen and generated a new acc. But the id.json I got from it is an array of numbers. Is this whats referred to as Machine readable? It didn't look like a typical json file

**Comment by staccDOTsol** [comment]
yup! it's a uint8array that computers can read, just a decrypted privatekey

**Comment by staccDOTsol** [comment]
you can go pk <-> uint8array like this:

https://www.npmjs.com/package/bs58

**Comment by fxtrader-react** [comment]
Ok thanks for help. Will get it up on running now with this help

**Comment by fxtrader-react** [comment]
I've been following the project ever since and I can understand it's still a work in progress. I see from your update that you've had more success with the arbitrage. Only issue is it's still a bit complicated and the read me doesn't help much. I noticed you changed the code a bit and now and the var "meAndTheBoys" in mango.simple.client.ts bugs out as you changed the name to "meAndThePeeps".
I think the python files are most important but the read me doesn't really point out to start from there. Also, I got the docker to build when i put idd.json in the mango-service-v3 and it seemed to work but no trades on delegated account in mango so I know I'm still missing something. A better more straight forward readme will help a lot.

**Comment by staccDOTsol** [comment]
I agree, to be honest I had created a much better readme at the request of another party, but I abandoned those commits to start from scratch as I couldn't get the thing to run either.

I think a substantially more valuable proposition would be this user experience flow:

1. you find the repo and/or @STACCart pinned tweet
2. you create a new Mango account using a referral link I provide
3. this allows Jare to earn 16% perpetual fees, along with the user saving an additional 4% as discount
4. user delegates to Jare via Mango
5. Jare adds the user's address to meAndThePeeps
6. the codebase will be rewritten in the following ways:
a. disaster preparedness, ie. taker orders to avoid certain unhedged or unbalanced pitfalls
b. taker orders in order to capture more opportunity faster, using a mechanism that ways the %APY vs the fees to get into this position quicker
c.  some sysadmin stuff to ensure the thing runs 24/7

I believe this is a much less confusing implementation that would yield. One downside is that I would need to cover the sol network fees, but in recent iterations I have reduced this cost significantly

Would  you be interested in this arrangement @fxtrader-react ?

**Comment by fxtrader-react** [comment]
This sounds like a solid idea. plus you get directly rewarded for your effort. How, when and where do we get started. Maybe i keep checking your twitter for updates on this? Also since monitoring the bot what is the success of it. Were you successful with implementing the arbitrage technique.

**Comment by staccDOTsol** [comment]
Cheers, I can update you here in this ticket too :)

I should note a couple things:

1. Mango 'account' page sometimes has not-quite-right historical balance data
2. this thing has only run a couple days - it is a very small sample size
3. historical performance is no indication of future results

**Comment by fxtrader-react** [comment]
Yup, I agree with all three points. This is all speculation at the end of the day. I will be keeping track here and your twitter for further updates. When can I expect to get started?

**Comment by staccDOTsol** [comment]
Soon as any of my generous benefactors get me 10k MNGO ;) is the requirement for the referral bits.

hey check this out, we should be able to trustlessly pool with this:

https://twitter.com/Toxic_Mangooo/status/1508638362634448898

I guess I can probably take a look at their repo

**Comment by staccDOTsol** [comment]
yea I forked both repos.

I guess they have their program on devnet

**Comment by staccDOTsol** [comment]
@fxtrader-react i spent the day dockerizing literally everything,

it's now much easier to run, but it's currently in devnet mode ;)

it's running a single account as a pool on devnet, check https;//riptide.fair3d.me (can get USDC from devnet.mango.markets)
